src/Explorer.js

```diff
@@ -1276,9 +1759,94 @@ class FieldView extends React.PureComponent<FieldViewProps, {}> {
           </span>
+          {!this.state.displayFieldActions ? null : (
+            <button
+              type="submit"
+              className="toolbar-button"
+              onClick={event => {
+                event.preventDefault();
+                event.stopPropagation();
+                // 1. Create a fragment spread node
+                // 2. Copy selections from this object to fragment
+                // 3. Replace selections in this object with fragment spread
+                // 4. Add fragment to document
+                const typeName = type.name;
+                let newFragmentName = `${typeName}Fragment`;
+
+                const conflictingNameCount = (applicableFragments || []).filter(
+                  (fragment: FragmentDefinitionNode) => {
+                    return fragment.name.value.startsWith(newFragmentName);
+                  },
+                ).length;
+
+                if (conflictingNameCount > 0) {
+                  newFragmentName = `${newFragmentName}${conflictingNameCount}`;
+                }
+
+                const childSelections = selection
+                  ? selection.selectionSet
+                    ? selection.selectionSet.selections
+                    : []
+                  : [];
+
+                const nextSelections = [
+                  {
+                    kind: 'FragmentSpread',
+                    name: {
+                      kind: 'Name',
+                      value: newFragmentName,
+                    },
+                    directives: [],
+                  },
+                ];
+
+                const newFragmentDefinition: FragmentDefinitionNode = {
+                  kind: 'FragmentDefinition',
+                  name: {
+                    kind: 'Name',
+                    value: newFragmentName,
+                  },
+                  typeCondition: {
+                    kind: 'NamedType',
+                    name: {
+                      kind: 'Name',
+                      value: type.name,
+                    },
+                  },
+                  directives: [],
+                  selectionSet: {
+                    kind: 'SelectionSet',
+                    selections: childSelections,
+                  },
+                };
+
+                const newDoc = this._modifyChildSelections(
+                  nextSelections,
+                  false,
+                );
+
+                if (newDoc) {
+                  const newDocWithFragment = {
+                    ...newDoc,
+                    definitions: [...newDoc.definitions, newFragmentDefinition],
+                  };
+
+                  this.props.onCommit(newDocWithFragment);
+                } else {
+                  console.warn('Unable to complete extractFragment operation');
+                }
+              }}
+              style={{
+                ...styleConfig.styles.actionButtonStyle,
+              }}>
+              <span>{'?'}</span>
```

Comment:
This needs a different icon from '?'. Maybe '...' ?

Can you also give it a `title` prop that describes what it will do?